### PR TITLE
Laboratory work #3, Matuschak Anastasia - 20FPL1

### PR DESCRIPTION
**Pull request by almatuschak**

### CODE REVIEW

File: lab_3/main.py [hunk 1]
```diff
@@ -85,7 +153,20 @@ def decode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
...
+    decoded_text = ()
+    for sentence in corpus:
+        word_tuple = ()
+        for word in sentence:
+            letter_tuple = ()
+            for letter_id in word:
+                letter_tuple += (storage.get_id_by_letter(letter_id),)
+            word_tuple += (letter_tuple,)
+        decoded_text += (word_tuple,)
+    return decoded_text
```
Comment (almatuschak) [inline lab_3/main.py:169]: this function doesn't pass the "ideal" test and i cannot understand why

Comment (WhiteJaeger) [reply lab_3/main.py:169]: You have to get the letter by id as it is a decoding function. Now it is `get_id_by_letter` but there should be used the `get_letter_by_id` method.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/main.py [hunk 2]
```diff
@@ -222,7 +355,25 @@ def save(self, name: str) -> int:
...
+                for element in n_gram:
+                    word += self.storage.get_letter_by_id(element)
+                freq[word] = frequency
+                word = ''
+        profile_as_dict['freq'] = freq
+        profile_as_dict['n_words'] = self.n_words
+        profile_as_dict['name'] = self.language
+        with open(name, 'w', encoding='UTF-8') as file:
+            json_string = json.dumps(profile_as_dict)
+            file.write(json_string)
```
Comment (WhiteJaeger) [inline lab_3/main.py:374]: There is the `.dump` method in the `json` package that combines these two lines.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_3/main.py (hunk 2 above)
Comment (almatuschak) [reply lab_3/main.py]: Done:
json.dump(profile_as_dict, file)

**Pull request closed without merging**